src/error.rs

```diff
@@ -1,26 +1,107 @@
+//! Enumerates all errors that this crate may return.
+//!
+//! [Error] is the top level error enum.
```

Comment:
The convention for references to code identifiers is to make them be in code-font:
```suggestion
//! [`Error`] is the top level error enum.
```